GenericsHomeWork/Node.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace GenericsHomeWork;
+
+public class NodeCollection<T> : ICollection<T>
+{
+    public T Value { get; set; }
+    public NodeCollection<T> Next { get; private set; }
+
+    public NodeCollection(T value)
+    {
+        Value = value;
+        Next = this;
+    }
+
+    public override string ToString()
+    {
+        return Value?.ToString() ?? string.Empty;
+    }
+
+    public void Add(T item)
+    {
+        Append(item);
+    }
+
+    public bool Contains(T item)
+    {
+        return Exists(item);
+    }
+
+    public void Append(T value)
+    {
+        if (Exists(value))
+        {
+            throw new InvalidOperationException($"Value '{value}' already exists in the list.");
```

Comment:
I think the exception type should be changed from `InvalidOperationException` to `ArgumentException`, because the use-case for the InvalidOperation exception is when the current state of an object makes the operation invalid, not that the operation isn't allowed to happen. 

Also, a bit of a nitpick, formatting the exception with `nameof(value)` being passed after the message is more common and lets you describe the 'no duplicates allowed', versus just alerting about the duplicate. 

```suggestion
            throw new ArgumentException("No duplicate values allowed in the list.", nameof(value));
```

Reply:
I changed it, thank you for your suggestion!